reading_scraper.py

```diff
@@ -29,115 +53,173 @@ def useragent_generator():
 
-        # if domain is 'special' or if title is blank, try cached version
-        if (reading.title == '' 
-            or '403 Forbidden' in reading.title 
-            or any(domain in url for domain in special_sites)):
-            # print('needing to cache')
-            # update header to refer from google
-            full_headers = {
-                'Accept': '*/*',
-                'Accept-Encoding': 'gzip, deflate, br',
-                'Accept-Language': 'en-US,en;q=0.9',
-                'content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-                'Cache-Control': 'no-cache',
-                'DNT': '1',
-                'Pragma': 'no-cache',
-                'Referer': 'https://www.google.com/',
-                'User-Agent': useragent_generator(),
-                'Sec-Fetch-Dest': 'document',
-                'Sec-Fetch-Mode': 'navigate',
-                'Sec-Fetch-Site': 'cross-site',
-                'Upgrade-Insecure-Requests': '1',
-            }
+    # if domain is 'special' or if title is blank, try cached version
+    if (reading == '' or reading.title == '' 
+        or '403 Forbidden' in reading.title
+        or "Bloomberg" in reading.title): 
+        # or any(domain in url for domain in special_sites)):
+        print('needing to cache')
+        # update header to refer from google
+        full_headers = {
+            'Accept': '*/*',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Accept-Language': 'en-US,en;q=0.9',
+            'content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            'Cache-Control': 'no-cache',
+            'DNT': '1',
+            'Pragma': 'no-cache',
+            'Referer': 'https://www.google.com/',
+            'User-Agent': useragent_generator(),
+            'Sec-Fetch-Dest': 'document',
+            'Sec-Fetch-Mode': 'navigate',
+            'Sec-Fetch-Site': 'cross-site',
+            'Upgrade-Insecure-Requests': '1',
+        }
+        try:
             reading = g.extract(url=cached_url)
-            # print(reading.title)
-        
-        #  if we cache and 404 error is thrown, back to normal
-        if ('Error 404 (Not Found)' in reading.title or 'https:///search?q=cache:' in reading.title):
-            # print('using link preview')
-            r = requests.get(f'http://api.linkpreview.net/?key={link_preview}&q={url}')
-            j = r.json()
-            title = j['title']
-            description = j['description']
-            image = j['image']
-    except:
-        reading = 'None'
+        except:
+            print("Error occurred with 1st try getting reading data: ", sys.exc_info())
+            reading = ''
+        # print(reading.title)
+    
+    #  if we cache and 404 error is thrown, back to normal
+    if ('Not Found' in reading.title 
+        or str(reading.title).startswith("https:///search?q=cache:")
+        or '404' in reading.title
+        or "Bloomberg" in reading.title
```

Comment:
See comment on line 98. I think it will be good to place these title conditionals in their own function so they're consistent across the file

Reply:
I think this would be easier as part of refactoring the code into a class instead of a string of functions. That was going to be the next step of this for me.